docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+# Sidechains
+
+## Overview
+
+This document defines the ledger objects, transactions, RPC commands, and
+additional servers that are used to support sidechains on the XRP ledger. A
+sidechain is an independent ledger. It has its own validators and can have its
+own set of custom transactions. Importantly, there is a way to move assets from
+the mainchain to the sidechain and there is a way to return those assets from
+the sidechain to the mainchain. This key operation is called a cross-chain
+transfer. A cross-chain transfer is not a single transaction. It happens on two
+chains, requires multiple transactions and involves an additional server type
+called a "witness". This key operation is described in a section below.
+
+After the mechanics of a cross-chain transfer are understood, an overview of
+some supporting transactions are described. This includes transactions to create
+sidechains, add signatures, create accounts across chains, and change sidechain
+parameters.
+
+The next sections describe the witness server, how to set-up a sidechain, error
+handling, and trust assumptions.
+
+Finally, a detailed reference for the new ledger objects and transactions are
+presented as a reference.
+
+## Nomenclature
+
+Cross-chain transfer: A transaction that moves assets from the mainchain to the
+sidechain, or returns those assets from the sidechain to the mainchain
+
+Cross-chain sequence number: A ledger object used to prove ownership of the
```

Comment:
@seelabs Probably a dumb question but does the sequence number (whether an unsigned int or a unsigned long int) rollover at some point? In theory can a malicious actor do a replay attack when sequence number rolls over to 0 at some point.

Reply:
No, it doesn't roll over, once you reach the max value you can send no more transactions (or we'll have to add an amendment to fix this). However, this is a very large number. For all practical purposes we'll never run out (sequence number on accounts have the same property.)

Reply:
Sounds good, thanks